### DISCUSSION TITLE
`searchurls` and character substitution

### DISCUSSION THREAD

**Discussion by sludgeFudge** [tridactyl/tridactyl#5110] [Q&A] [answered]
Hey, love this plugin. 

I am successfully able to set up the `searchurls` I want. However, with [Jisho.org](https://jisho.org/) I do need the `#` character substituted with `%23` for certain search operations.

This is my set up:
```
set searchurls.jisho https://jisho.org/search/%s
```

Normal searches work fine, but if, for example I want to look up the dedicated kanji page for "人", I must search for:
```
人 #kanji
```

With the current setup the url becomes:
```
https://jisho.org/search/%E4%BA%BA%20#kanji

(firefox simplified:)
https://jisho.org/search/人 #kanji
```

However, that results in the site ignoring the `kanji` tag.
To get the site to acknowledge the tag, the url must be:
```
https://jisho.org/search/%E4%BA%BA%20%23kanji

(firefox simplified:)
https://jisho.org/search/人 %23kanji
```

Is it possible to substitute `#` with `%23`?\
Thank you.

**Comment by Chic-Tweetz** [comment]
I don't know that there's a nice quick way to alter the text before searching, but you could include the %23 in the searchurl itself.

This wouldn't work if you want to search without a tag, so you'd need a separate searchurl for that, but for searches with tags you could use:
```
set searchurls.jisho https://jisho.org/search/%23%s
```
Then the equivalent search would be
```
kanji 人
```
Or (if you prefer the tag at the end)
```
set searchurls.jisho https://jisho.org/search/%s[:-1] %23%s[-1:]
```
for the search to be
```
人 kanji
```

**Comment by sludgeFudge** [comment]
Thanks for the reply. Having a separate `searchurl` is a good idea. I'll just use that for now.
I've discovered the `command`, `composite` and `urlmodify` commands. I'll keep experimenting for a bit and see if I can get what I want.

**Comment by Chic-Tweetz** [comment]
Good call, your original searchurl along with
```
autocmd DocStart ^http(s?)://jisho.org/search/ urlmodify -t # %23
```
seems to do the trick!

**Reply by sludgeFudge** [reply]
Yes, thank you! That works fantastically! 
I would like it if did the substitution before the page was loaded, but `urlmodify` only seems to change the current URL.
I did find the `autocmd` but I didn't realise that it just ran automatically so thank you for that.

If I may ask, does the `^` character represent the beginning of the URL?

**Reply by Chic-Tweetz** [reply]
Yes, `^` is the beginning, AutoCmd uses regex to check URLs. Though I just copied that format from the "New reddit is bad" line in the [exemplar .tridactylrc](https://github.com/tridactyl/tridactyl/blob/master/.tridactylrc) :)

I also copied DocStart from there, but there are other options that might modify the URL sooner. HOWEVER! @bovine3dom's suggestion is just what you were after to begin with so I'd go with that!

**Comment by bovine3dom** [comment, marked as answer]
I think this is a bug: search fragments should probably be URI encoded, so I've filed an issue https://github.com/tridactyl/tridactyl/issues/5122

In the meantime you can probably use jsurls (edited to fix the issue chic-tweetz pointed out):

```
set jsurls.jisho2 query => "https://jisho.org/search/" + encodeURIComponent(query)
```

(untested!)

**Reply by Chic-Tweetz** [reply]
Oh even better! I wasn't aware of jsurls, that's just the thing. This needs the "return" removed though! As in:
```
set jsurls.jisho2 query => "https://jisho.org/search/" + encodeURIComponent(query)
```

**Reply by sludgeFudge** [reply]
Oh this is awesome. Thanks you two.
Works as I want it to.
